packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/environmentVariableWalkthrough.ts

```diff
@@ -0,0 +1,161 @@
+  switch (operation) {
+    case 'add': {
+      const { newEnvironmentVariableKey, newEnvironmentVariableValue } = await addEnvironmentVariableQuestion(environmentVariables);
+      setEnvironmentVariable(context, resourceName, { [newEnvironmentVariableKey]: newEnvironmentVariableValue });
+      return await askEnvironmentVariableQuestions(context, resourceName);
+    }
+    case 'update': {
+      const { newEnvironmentVariableKey, newEnvironmentVariableValue, targetedKey } = await updateEnvironmentVariableQuestion(
+        environmentVariables,
+      );
+      deleteEnvironmentVariable(context, resourceName, targetedKey);
+      setEnvironmentVariable(context, resourceName, { [newEnvironmentVariableKey]: newEnvironmentVariableValue });
+      return await askEnvironmentVariableQuestions(context, resourceName);
+    }
+    case 'remove': {
+      const targetedKey = await removeEnvironmentVariableQuestion(environmentVariables);
+      deleteEnvironmentVariable(context, resourceName, targetedKey);
+      return await askEnvironmentVariableQuestions(context, resourceName);
+    }
+    case 'abort':
+    default:
+      const updatedEnvironmentVariables = getStoredEnvironmentVariables(context, resourceName);
+      return Promise.resolve({
+        environmentMap: Object.keys(updatedEnvironmentVariables).reduce(
+          (acc, cur) => ({
+            ...acc,
+            [cur]: { Ref: _.camelCase(cur) },
+          }),
+          {},
+        ),
+      });
+  }
+};
+
+const selectEnvironmentVariableQuestion = async (environmentVariables: object): Promise<'add' | 'update' | 'remove' | 'abort'> => {
+  const { operation } = await inquirer.prompt([
+    {
+      name: 'operation',
+      message: 'Select what you want to do with environment variables:',
+      type: 'list',
+      choices: [
+        {
+          value: 'add',
+          name: 'Add new environment variable',
+        },
+        {
+          value: 'update',
+          name: 'Update existing environment variables',
+          disabled: _.size(environmentVariables) === 0,
+        },
+        {
+          value: 'remove',
+          name: 'Remove existing environment variables',
+          disabled: _.size(environmentVariables) === 0,
+        },
+        {
+          value: 'abort',
+          name: "I'm done",
+        },
+      ],
+    },
+  ]);
+
+  return operation;
+};
+
+const addEnvironmentVariableQuestion = async (
+  environmentVariables: object,
+): Promise<{ newEnvironmentVariableKey: string; newEnvironmentVariableValue: string }> => {
+  const { newEnvironmentVariableKey, newEnvironmentVariableValue } = await inquirer.prompt([
+    {
+      name: 'newEnvironmentVariableKey',
+      message: 'Enter the environment variable name:',
+      type: 'input',
+      validate: input => {
+        if (input.match(/^[a-zA-Z0-9_]+$/) === null) {
+          return 'Use only /^[a-zA-Z0-9_]+$/';
+        }
+        if (Object.keys(environmentVariables).includes(input)) {
```

Comment:
I think you can use `Object.prototype.hasOwnProperty()` here.

Reply:
Yes! I update to _.has in this.